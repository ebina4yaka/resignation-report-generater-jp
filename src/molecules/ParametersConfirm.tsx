import React, { useContext, useEffect } from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import ExportDialog from './ExportDialog'
import { parametersContext } from '../context/useParameters'
import ConfirmButton from '../atoms/ConfirmButton'

const useStyles = makeStyles(
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      flexDirection: 'row',
      justifyContent: 'center',
    },
  })
)

const useIsButtonDisable = (): boolean => {
  const context = useContext(parametersContext)
  const { name, companyName, dateOfRetirement, reason } = context
  const [isDisable, setIsDisable] = React.useState(false)
  useEffect(() => {
    setIsDisable(
      name === '' ||
        companyName === '' ||
        dateOfRetirement === '' ||
        reason === ''
    )
  }, [context])
  return isDisable
}

const useDialogOpen = (): {
  isOpen: boolean
  handleOpen: () => void
  handleClose: () => void
} => {
  const [isOpen, setOpen] = React.useState(false)
  const handleOpen = () => {
    setOpen(true)
  }
  const handleClose = () => {
    setOpen(false)
  }
  return { isOpen, handleOpen, handleClose }
}

export default function ParametersConfirm(): React.ReactElement {
  const classes = useStyles()
  const { isOpen, handleOpen, handleClose } = useDialogOpen()
  return (
    <div className={classes.root}>
      <ConfirmButton
        text="確認"
        isDisable={useIsButtonDisable()}
        onClickFunction={handleOpen}
      />
      <ExportDialog isOpen={isOpen} handleClose={handleClose} />
    </div>
  )
}
