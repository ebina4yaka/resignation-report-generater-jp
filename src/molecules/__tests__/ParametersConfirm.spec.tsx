import React from 'react'
import renderer from 'react-test-renderer'
import ParametersConfirm from '../ParametersConfirm'

jest.mock('../../atoms/ConfirmButton', () => 'MockedConfirmButton')
jest.mock('../ExportDialog', () => 'MockedExportDialog')

test('Snapshot test for ParametersConfirm', () => {
  const tree = renderer.create(<ParametersConfirm />).toJSON()
  expect(tree).toMatchSnapshot()
})
