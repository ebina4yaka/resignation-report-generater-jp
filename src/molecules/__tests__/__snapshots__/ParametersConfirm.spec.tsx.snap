// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for ParametersConfirm 1`] = `
<div
  className="makeStyles-root-1"
>
  <MockedConfirmButton
    isDisable={false}
    onClickFunction={[Function]}
    text="確認"
  />
  <MockedExportDialog
    handleClose={[Function]}
    isOpen={false}
  />
</div>
`;
